### ISSUE TITLE
ParseError no longer use the error.cause property

### ISSUE THREAD

**Issue by tusharf5** [aws-powertools/powertools-lambda-typescript#2773]
### Expected Behavior

Earlier the error thrown within the handler would be attached to the ParseError to the .cause property. This no longer is the case. This broke one of our prod services that were relying on this property.

I think it was unintentionally removed as part of dropping Node 16 support.

Here's the link to the change.

https://github.com/aws-powertools/powertools-lambda-typescript/commit/e4eee73770ffccead9212a566335ec256a31af7d#diff-27fac689aa83bd57991ab46c30f4cd72584c529baa433ce899e8b7bd6db7a7deL21

### Current Behavior

ParseError.cause is undefined

### Code snippet

Just throw any validation error in the handler and look for error.cause property.

### Steps to Reproduce

You can run any tests in this file "packages/parser/tests/unit/envelope.test.ts" and check for error.cause property. It will be undefined.

### Possible Solution

I've submitted a PR which reverts the ParseError change that removed this behaviour. 

### Powertools for AWS Lambda (TypeScript) version

latest

### AWS Lambda function runtime

20.x

### Packaging format used

npm

**Comment by dreamorosi** [comment]
Hi @tusharf5, thank you for opening this issue.

You are right in that the `cause` property was removed unintentionally; the change was made because since dropping Node.js 16 we can now safely assume that [all errors have the `cause` property by default](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause).

The idea was to use the "native" property instead of one defined by us, however it seems that I introduced a bug by not propagating the second argument we pass when we throw a `ParseError` to its parent class [here](https://github.com/aws-powertools/powertools-lambda-typescript/blob/e648f9b0a43182449bcd0847d6d078d81a6233e3/packages/parser/src/errors.ts#L10). The error should've been like this:

```diff
class ParseError extends Error {
  public constructor(message: string, options?: { cause?: Error }) {
    const errorMessage = options?.cause
      ? `${message}. This error was caused by: ${options?.cause.message}.`
      : message;
-   super(errorMessage);
+   super(errorMessage, options);
    this.name = 'ParseError';
  }
}
```
 

Even though the API for this specific utility is still to be considered unstable during its [beta period](https://github.com/aws-powertools/powertools-lambda-typescript/blob/e648f9b0a43182449bcd0847d6d078d81a6233e3/packages/parser/README.md?plain=1#L4-L6), we should have included this property in our tests.

--

Thank you for submitting the PR to fix it, I've left a couple comments. Once addressed it's good to be merged.
